subsys/console/tty.c

```diff
@@ -149,12 +160,54 @@ static int tty_getchar(struct tty_serial *tty)
 	return c;
 }
 
+static ssize_t tty_read_unbuf(struct tty_serial *tty, void *buf, size_t size)
+{
+	u8_t *p = buf;
+	size_t out_size = 0;
+	int res = 0;
+	u32_t timeout = tty->rx_timeout;
+
+	while (size) {
+		u8_t c;
+		res = uart_poll_in(tty->uart_dev, &c);
+		if (res <= -2) {
+			/* Error occured, best we can do is to return
+			 * accumulated data w/o error, or return error
+			 * directly if none.
+			 */
+			if (out_size == 0) {
+				errno = res;
+				return -1;
+			}
+			break;
+		}
+
+		if (res == 0) {
+			*p++ = c;
+			out_size++;
+			size--;
+		}
+
+		if (size == 0 || (timeout != K_FOREVER && timeout-- == 0)) {
+			break;
+		}
+
+		k_sleep(1);
```

Comment:
@nordic-krch :

> this triggers rescheduling. If blocking tty is used in panic then that should not happen. Especially that call will most likely happen from irq (e.g. hardfault) and k_sleep is for thread only (asserts in irq). k_busy_wait would be better option.

Did you notice that this k_sleep() is in read() function? You're not interested/not going to call read() for logging, right? Otherwise the idea of k_sleep() is exactly to avoid burning CPU at 100% usage while waiting for UART input in the not interrupt driven case, letting it sleep or process other tasks for a bit.